### PR TITLE
upgrade to babel 6

### PR DESCRIPTION
**Pull request by kidwm**
Also, remove unused sass-loader and problematic react-burger-menu

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,13 +42,11 @@
     "json-loader": "^0.5.3",
     "lodash": "^3.10.1",
     "mobile-detect": "^1.3.0",
-    "node-sass": "^3.3.3",
     "nodemon": "^1.6.0",
     "normalizr": "^1.0.0",
     "qs": "^5.2.0",
     "react": "^0.14.3",
     "react-addons-test-utils": "^0.14.0",
-    "react-burger-menu": "^1.4.4",
```
Comment (nickhsine) [inline package.json]: we will need ```react-burger-menu``` for navigation bar.

Comment (kidwm) [inline package.json]: You could write your own component to replace this.

Comment (nickhsine) [inline package.json]: Why don't we just use this?

File: webpack.config.js [hunk 2]
```diff
@@ -12,11 +12,11 @@ const config = {
   module: {
     loaders: [
       {
-        test: /\.js$/,
+        test: /\.jsx?$/,
```
Comment (nickhsine) [inline webpack.config.js]: We are writing es6, so we won't have jsx file

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import 'babel-core/polyfill'
+require('babel-core/register')
```
Comment (nickhsine) [inline src/index.js]: Use import?

File: webpack.config.js (hunk 2 above)
Comment (kidwm) [inline webpack.config.js]: I just merged these two original loaders.

BTW, your so-said es6 file in this repo are all written in jsx and not pure ES2015.

File: src/index.js (hunk 3 above)
Comment (kidwm) [inline src/index.js]: this is not actual module and should be more friendly to node in server side.

File: package.json (hunk 1 above)
Comment (kidwm) [inline package.json]: It required babel 5 in its dependency.

File: webpack.config.js (hunk 2 above)
Comment (nickhsine) [inline webpack.config.js]: Got it, didn't see ```?``` at first time.

File: package.json [hunk 4]
```diff
@@ -5,8 +5,9 @@
   "scripts": {
     "start": "webpack --progress && node browsersync-server",
     "production": "webpack -p --progress && forever start server",
-    "test": "NODE_ENV=test NODE_PATH=. mocha --compilers js:babel/register -r spec/support/setup.mocha.js --recursive 'spec/**/*.test.js' -w",
-    "lint": "eslint ."
+    "test": "NODE_ENV=test NODE_PATH=. mocha --compilers js:babel-core/register -r spec/support/setup.mocha.js --recursive 'spec/**/*.test.js' -w",
+    "lint": "eslint .",
+    "validate": "npm ls"
```
Comment (kaddopur) [inline package.json]: should we add `validate` to `pre-commit`?
https://github.com/kidwm/twreporter-react/blob/babel-6/package.json#L96

Comment (kidwm) [inline package.json]: It was auto added by git-validate, although I really very hate what it has done.

**Comment by nickhsine** [comment]
@kidwm
因為我們最近要上目錄頁了，
所以必須先用```react-burger-menu```。
所以這個PR可能要等到我們之後implement自己的menu後再merge。
謝謝你的幫忙

**Comment by kidwm** [comment]
That's fine. I'll rearrange this PR after then.

**Comment by nickhsine** [comment]
@kidwm 
我剛剛試了一下你的PR，
同時也保留 ```react-burger-menu```，
但似乎沒有什麼問題噎。
可以說明一下你遇到什麼問題嗎？

**Comment by kidwm** [comment]
@nickhsine if you use npm 3, `react-burger-menu` will introduce babel 5 and force others use the wrong version.

**Comment by nickhsine** [comment]
@kidwm 你用node 5 and npm 3嗎？

**Comment by kidwm** [comment]
@nickhsine yes, frontend engineer live in the frontier.

**Comment by nickhsine** [comment]
Cool.
We will add menu made by ourselves asap.

**Comment by nickhsine** [comment]
@kidwm 
we just use node@5.0.0 and npm@3.3.6,
and it works normally. 
I haven't encounter the babel problem.

**Comment by kidwm** [comment]
Done.
and you can see the result of `npm ls react-burger-menu`

**Comment by nickhsine** [comment]
@kidwm Have you passed lint?

**Comment by kidwm** [comment]
no, I have no time to cater that

nick <notifications@github.com> 於 2015年12月24日 星期四寫道：

> @kidwm <https://github.com/kidwm> Have you passed lint?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twreporter/twreporter-react/pull/99#issuecomment-167033148>
> .
>


-- 
If you don't stand for something, you'll fall for anything.

http://kidwm.net/

**Comment by nickhsine** [comment]
@kidwm 
ok, let us do that.

**Pull request merged**